src/bson.c

```diff
@@ -617,10 +621,22 @@ void object_to_bson(zval *object, php_phongo_bson_flags_t flags, const char *key
 				return;
 			}
 
-			if (Z_TYPE_P(obj_data) != IS_ARRAY) {
-				phongo_throw_exception(PHONGO_ERROR_UNEXPECTED_VALUE TSRMLS_CC, "Expected %s() to return an array, %s given", BSON_SERIALIZE_FUNC_NAME, zend_get_type_by_const(Z_TYPE_P(obj_data)));
+			if (Z_TYPE_P(obj_data) != IS_ARRAY && !(Z_TYPE_P(obj_data) == IS_OBJECT && instanceof_function(Z_OBJCE_P(obj_data), zend_standard_class_def TSRMLS_CC))) {
+				const char *classname = NULL;
+				zend_uint classname_len = 0;
+				int classname_free = 0;
+
+				if (Z_TYPE_P(obj_data) == IS_OBJECT) {
+					classname_free = !zend_get_object_classname(obj_data, &classname, &classname_len TSRMLS_CC);
```

Comment:
`zend_get_object_classname` has been removed in PHP-7.
Please continue to use `Z_OBJCE_P(object)->name`